javatests/arcs/core/storage/StorageProxyTest.kt

```diff
@@ -0,0 +1,133 @@
+package arcs.core.storage
+
+import arcs.core.crdt.CrdtData
+import arcs.core.crdt.CrdtModel
+import arcs.core.crdt.CrdtOperation
+import arcs.core.crdt.internal.VersionMap
+import com.google.common.truth.Truth.assertThat
+import com.nhaarman.mockitokotlin2.verifyNoMoreInteractions
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.verify
+import org.mockito.MockitoAnnotations
+
+/** Tests for [StorageProxy]. */
+@Suppress("UNCHECKED_CAST", "UNUSED_VARIABLE")
+@ExperimentalCoroutinesApi
+@RunWith(JUnit4::class)
+class StorageProxyTest {
```

Comment:
I think we need some tests here which try to cause deadlocks / test 'randomized' concurrent operations

Reply:
Agreed, I'll work on it

Reply:
I've upgraded AtomicFu so we can use their multi-platform ReentrantLock. That should mitigate the deadlock possibilities. I very much want some awesome randomized tests, but I'd like to do it as a follow-up CL integration test covering at minimum the collection handles + storage proxy + concurrent read/writes to said collection, the store fake (or even better a ram store).

Reply:
SGTM